plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -103,19 +152,55 @@ def __cmp__(self, other):
 
 
 class Distribution(Package):
-    UNIT_KEY_NAMES = ('id', 'family', 'variant', 'version', 'arch')
-    TYPE = ids.TYPE_ID_DISTRO
-
-    def __init__(self, family, variant, version, arch, metadata, id=None):
-        kwargs = locals()
-        # I don't know why this is the "id", but am following the pattern of the
-        # original importer
-        if kwargs['id'] is None:
+    # TODO add docstring to this class
+
+    distribution_id = mongoengine.StringField(required=True)
+    family = mongoengine.StringField(required=True)
+    variant = mongoengine.StringField(required=True)
+    version = mongoengine.StringField(required=True)
+    arch = mongoengine.StringField(required=True)
+
+    files = mongoengine.ListField()
+    timestamp = mongoengine.FloatField()
+    packagedir = mongoengine.StringField()
+
+    # Pretty sure the version_sort_index is never used for Distribution units
+    version_sort_index = mongoengine.StringField()
+
+    # For backward compatibility
+    _ns = mongoengine.StringField(default='units_distribution')
```

Comment:
If moved to after `meta`, this could be:
```
_ns = mongoengine.StringField(default=meta['collection'])
```
to eliminate the possibility of mismatch.

In fact, it seems this could be done in for all model classes when loaded.